copylot/gui/_qt/photom_control/helper_functions/draw_pattern_unit.py

```diff
@@ -0,0 +1,314 @@
+import math
+
+from PyQt5.QtCore import Qt
+from PyQt5.QtWidgets import (
+    QLabel,
+    QPushButton,
+    QButtonGroup,
+    QRadioButton,
+    QLineEdit,
+    QGroupBox,
+    QGridLayout,
+    QProgressBar,
+    QApplication,
+)
+
+from dac_controller.scan import DACscan
+from widgets.scan_algrthm.scan_algorithm import ScanAlgorithm
+
+
+class DrawPatternUnit(QGroupBox):
```

Comment:
these seem  highly re-usable, feel  free to  store helper widgets in  custom_widgets